src/scattering.F90

```diff
@@ -40,52 +41,68 @@ subroutine nu_scatter_elastic_e_total(neutrino_energy,type,lepton,eos_variables,
      stop "nu_scatter_elastic_e_total: lepton number weird...."
   endif
 
-  if (type.eq.1) then
-     crosssection = 3.0d0/8.0d0*sigma0/m_e**2*neutrino_energy* &
-          (eos_variables(tempindex)+eos_variables(mueindex)/4.0d0)* &
-          ((c_v_prime+c_a_prime)**2+(c_v_prime-c_a_prime)**2/3.0d0)
-  else
-     stop "no transport crosssection?"
+  !ONLY TRANSPORT OPACITY IMPLEMENTED
+  !if (transport.eq.1) then
+  crosssection = 3.0d0/8.0d0*sigma0/m_e**2*neutrino_energy* &
+       (eos_variables(tempindex)+eos_variables(mueindex)/4.0d0)* &
+       ((c_v_prime+c_a_prime)**2+(c_v_prime-c_a_prime)**2/3.0d0)
+  delta = 0
+  !else
+  !   stop "no transport crosssection?"
+  !endif
+
+  ! check output
+  if(delta<-1.0d0 .or. delta>1.0d0) then
+     write(*,*) "Invalid value of delta: ",delta
+     stop
   endif
-
-
+     
 end subroutine nu_scatter_elastic_e_total
 
-subroutine nu_scatter_elastic_alpha_total(neutrino_energy,type,lepton,crosssection)
+subroutine nu_scatter_elastic_alpha_total(neutrino_energy,transport,lepton,crosssection,delta)
 
   use nulib
   implicit none
   
   !inputs
   real*8, intent(in) :: neutrino_energy !in MeV
-  integer, intent(in) :: type !transport (type.eq1) or absorption (type.eq.0) crosssection
+  integer, intent(in) :: transport !transport (transport.eq1) or absorption (transport.eq.0) crosssection
   integer, intent(in) :: lepton !lepton number, >0 for neutrino,  <0 for antineutrino
 
   !output
   real*8,intent(out) :: crosssection !in cm^2
-
+  real*8,intent(out) :: delta !dimensionless
+  
   crosssection = 4.0d0*sigma0*(neutrino_energy/m_e)**2*sin2thetaW**2
 
-  !transport crosssection
-  if (type.eq.1) then
-     crosssection = 2.0d0/3.0d0*crosssection
-  end if
+  !ONLY TRANSPORT OPACITY IMPLEMENTED
```

Comment:
same here as for the CE\nuNS, delta = 1 and the non-transport cross section is 3/2 of the transport one

Reply:
Done.